src/main/java/org/cactoos/bytes/HexBytes.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.bytes;
+
+import java.io.IOException;
+import org.cactoos.Bytes;
+import org.cactoos.Text;
+import org.cactoos.text.FormattedText;
+
+/**
+ * Decodes origin {@link Text} using the hexadecimal encoding scheme.
+ *
+ * @author Alexander Menshikov (sharplermc@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+public final class HexBytes implements Bytes {
+    /**
+     * Origin hexadecimal text.
+     */
+    private final Text origin;
+
+    /**
+     * Ctor.
+     *
+     * @param origin Hexadecimal text.
+     */
+    public HexBytes(final Text origin) {
+        this.origin = origin;
+    }
+
+    @Override
+    public byte[] asBytes() throws IOException {
+        final String hex = this.origin.asString();
+        if ((hex.length() & 1) == 1) {
+            throw new IOException("Length of hexadecimal text is odd");
+        }
+        final byte[] result = new byte[hex.length() / 2];
+        int index = 0;
+        while (index < hex.length()) {
+            final int most = this.checkedDigit(hex.charAt(index));
+            final int less = this.checkedDigit(hex.charAt(index + 1));
+            result[index >>> 1] = (byte) ((most << 4) + less);
+            index += 2;
+        }
+        return result;
+    }
+
+    /**
+     * Wrap around {@link Character#digit(char, int)}.
+     *
+     * @param character Hexadecimal character.
+     * @return Numeric Value of the hex character.
+     * @throws IOException If fails.
+     */
+    private int checkedDigit(final char character) throws IOException {
```

Comment:
Removing keyword `static` doesn't change the semantic of this method whatsoever. :) It is still an awkward utility **procedure** that has no [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)) relative to the `HexBytes` class.

It can be way better than that. You can turn this method into an **Object**. In fact, you even named this method as an object: `CheckedDigit`. This object is definitely a [Scalar](https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/Scalar.java). You may even want to extend [NumberEnvelope](https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/scalar/NumberEnvelope.java) to make that scalar easily convertable between different types of numbers. Just look up the code base, there are plenty of examples out there (e.g. [LengthOf](https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/io/LengthOf.java)).

Reply:
I'm not sure about `NumberEnvelope` which wraps `Double`. Because `Character.digit` returned value is between `[-1:35]`. See the `Character.MAX_RADIX`.
I know the performance is not in first place, but it's a bit crazy to create an object for every char in `String`.
Maybe better to use `Func<Char, Integer>` which created with specific `radix`?

Please look at options and choose better:

Your option with `Scalar`:
```Java
final String hex = this.origin.asString();
        if ((hex.length() & 1) == 1) {
            throw new IOException("Length of hexadecimal text is odd");
        }
        final byte[] result = new byte[hex.length() / 2];
        int index = 0;
        while (index < hex.length()) {
            final int most = new CheckedDigit(
                hex.charAt(index),
                16
            ).value();
            final int less = new CheckedDigit(
                hex.charAt(index),
                16
            ).value();
            result[index >>> 1] = (byte) ((most << 4) + less);
            index += 2;
        }
        return result;
```

My option with `Func`:

```Java
final String hex = this.origin.asString();
        if ((hex.length() & 1) == 1) {
            throw new IOException("Length of hexadecimal text is odd");
        }
        final byte[] result = new byte[hex.length() / 2];
        int index = 0;
        final CheckedDigit digit = new CheckedDigit(16);
        while (index < hex.length()) {
            final int most = digit(hex.charAt(index));
            final int less = digit(hex.charAt(index + 1));
            result[index >>> 1] = (byte) ((most << 4) + less);
            index += 2;
        }
        return result;
```

Reply:
> I'm not sure about `NumberEnvelope` which wraps Double. Because `Character.digit` returned value is between `[-1:35]`.

Yeah, I agree, looks like an overkill for this paticular case.

> Maybe better to use `Func<Char, Integer>` which created with specific `radix`?

Look at the description of the project. We are developing "Object-Oriented Java primitives". So there should be no doubt in choosing between an Object and a function. :)

Ok, let's see what you can do. You can turn your `hex` String into an iterable object and then map that iterable to a series of checked digits:

```java
checkedDigits = new Mapped<>(
  x -> { /* the body of checkedDigit procedure */ },
  new IterableOf<>(hex.toCharArray())
)
```

And instead of iterating `hex` you iterate `checkedDigits`. Does this look faster to you?

And using [IterableEnvelope](https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/iterable/IterableEnvelope.java) you can turned that entire expression into a separate object. I don't know how to call it though... `CheckedDigits`?